x-pack/platform/plugins/shared/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -492,6 +492,10 @@ export class ObservabilityAIAssistantClient {
       functionCalling: (simulateFunctionCalling ? 'simulated' : 'auto') as FunctionCallingMode,
     };
 
+    this.dependencies.logger.debug(
+      `Calling inference client with for name: "${name}" with options: ${JSON.stringify(options)}`
```

Comment:
can you use the callback signature here, to prevent JSON.stringify when the logger is not at debug level?